kubernetes/resource_kubernetes_service_test.go

```diff
@@ -811,6 +811,39 @@ func testAccCheckServicePorts(svc *api.Service, expected []api.ServicePort) reso
 	}
 }
 
+func testAccCheckloadBalancerIngressCheck(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %s", resourceName)
+		}
+
+		node, err := getFirstNode()
+
+		if err != nil {
+			return err
+		}
+
+		labels := node.GetLabels()
+
+		lb := "status.0.load_balancer.0.ingress.0"
+
+		if _, ok := labels["cloud.google.com/gke-nodepool"]; ok {
```

Comment:
I'm not sure I would even be as clever as this. I would just check if the `hostname` is there, if it's not there then try `ip` and if that's not there then surface an error.

Reply:
Yeah there were a couple of different ways to have checked for the `hostname` and the `ip`, one of them was just checking for `hostname` like you said but was also brought up that it would be great to check for the specific GKE label also by @alexsomesan. Had both planned and ready for PR but curious as to which approach to go with.